src/platform_impl/macos/app_state.rs

```diff
@@ -352,7 +352,14 @@ impl AppState {
     }
 
     pub fn handle_redraw(window_id: WindowId) {
-        HANDLER.handle_nonuser_event(EventWrapper::StaticEvent(Event::RedrawRequested(window_id)));
+        // Redraw request might come out of order from the OS.
+        // -> Don't go back into the callback when our callstack originates from there
+        if !HANDLER.get_in_callback() {
+            HANDLER.set_in_callback(true);
```

Comment:
That's racy. you should do either atomic `swap` or put the code into the critical section.

Reply:
oh and such an obvious one as well 😳 . Thanks!